File: SQL Connection/Purchase.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Statement;
import java.sql.PreparedStatement;



public class Purchase {
	
    public static void main(String[] args) {
      Connection connect = null;
  	  Statement statement = null;
  	  ResultSet resultSet = null;
  	  try{
    	Class.forName("com.mysql.jdbc.Driver");
  	  }
  	 catch (Exception ex) {
         
     }
        // Setup the connection with the DB
  	  try{
  		connect = DriverManager
  	            .getConnection("jdbc:mysql://localhost/inf551?"
  	                + "user=inf551&password=inf551");
  	// Statements allow to issue SQL queries to the database
        statement = connect.createStatement();
        // Result set get the result of the SQL query
        PreparedStatement ps = connect.prepareStatement("select distinct(product) from inf551.Purchase where buyer = ?");
        ps.setString(1,args[0]);
        resultSet = ps.executeQuery();
        int i =0;
        while (resultSet.next()) {
        	i =1;
    		String name = resultSet.getString(1);
    		System.out.println(name);
        }
        
        if( i == 0)
        	System.out.println("No data Found");
  	  }
  	catch (SQLException ex) {
	    // handle any errors
	    System.out.println("SQLException: " + ex.getMessage());
	    System.out.println("SQLState: " + ex.getSQLState());
	    System.out.println("VendorError: " + ex.getErrorCode());
	}     
    	
    }
}

File: SQL Connection/purchase.py
'''
Created on Mar 9, 2016

@author: yaminigoyal
'''
import sys
import mysql.connector
from mysql.connector import errorcode


try:
    cnx = mysql.connector.connect(user='inf551', password='inf551',
                              host='localhost',
                              database='inf551')
except mysql.connector.Error as err:
    if err.errno == errorcode.ER_ACCESS_DENIED_ERROR:
        print("Something is wrong with your user name or password")
    elif err.errno == errorcode.ER_BAD_DB_ERROR:
        print("Database does not exist")
    else:
        print(err)
else:
    cursor = cnx.cursor()
    buyer = sys.argv[1]
    query = ("SELECT distinct(product) FROM Purchase WHERE buyer =") + "\"" + buyer + "\""
    cursor.execute(query,buyer)
    i = 0
    for (product) in cursor:
        i = 1
        print(product)
    if i == 0:
        print "No data found"
    cursor.close()
    cnx.close()